apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/EndpointJsonComparatorBase.java

```diff
@@ -0,0 +1,163 @@
+
+/** Holds shared functionality between endpoint comparator tests in v1/v2. */
+public abstract class EndpointJsonComparatorBase extends ServerRequiredTest {
+
+  /** Test to use for an ignored field. */
+  protected static final String IGNORED_FIELD_TEXT = "IGNORED_FIELD";
+
+  /**
+   * Generates current endpoint response files, comparing them to the corresponding approved
+   * responses.
+   *
+   * <p>If you've recently made mapping changes, and this test fails, you may need to re-generate
+   * the expected response files by building the project with the maven parameter
+   * -DgenerateTestData=true which will run the test that generates new response files. This should
+   * only be done if you're sure your mapping changes are correct, as these files are used for
+   * regression testing future changes. Once generated, these newly minted files should be checked
+   * in with your mapping changes.
+   *
+   * @param endpointId the endpoint id
+   * @param endpointOperation the endpoint operation
+   */
+  @ParameterizedTest(name = "endpointId = {0}")
+  @MethodSource("data")
+  public void verifyCorrectEndpointResponse(String endpointId, Supplier<String> endpointOperation) {
+    String endpointResponse = endpointOperation.get();
+    assertJsonDiffIsEmpty(endpointId, endpointResponse);
+  }
+
+  /**
+   * Compares the approved and current responses for an endpoint.
+   *
+   * @param endpointId the name of the operation being tested, used to determine which files to
+   *     compare
+   * @param endpointResponse the endpoint response
+   */
+  public void assertJsonDiffIsEmpty(String endpointId, String endpointResponse) {
+    String expectedJson =
+        ServerTestUtils.readFile(
+            ServerTestUtils.generateEndpointJsonFileName(getExpectedJsonResponseDir(), endpointId));
+    AssertUtils.assertJsonEquals(
+        expectedJson, endpointResponse, ServerTestUtils.JSON_COMPARE_IGNORED_PATHS);
+  }
+
+  /**
+   * Gets the json response for the specified endpoint request string and default content type json
+   * header.
+   *
+   * @param requestString the request string
+   * @return the json response for
+   */
+  protected static String getJsonResponseFor(String requestString) {
+    return getJsonResponseFor(
+        requestString, new Headers(new Header("Content-Type", "application/json+fhir")));
+  }
+
+  /**
+   * Gets the json response for the specified endpoint request string and specified headers.
+   *
+   * @param requestString the request string
+   * @param headers the headers for the request
+   * @return the json response for
+   */
+  protected static String getJsonResponseFor(String requestString, Headers headers) {
+    return given()
+        .spec(requestAuth)
+        .headers(headers)
+        .expect()
+        .log()
+        .ifError()
+        .statusCode(200)
+        .when()
+        .get(requestString)
+        .then()
+        .extract()
+        .response()
+        .asString();
```

Comment:
Here's what all the tests use to call to the server; these ~11 lines are a lot simpler than the fhir server setup, json response interceptor setup + registration, and header registration that it used to do!